src/main/java/org/opensearch/neuralsearch/search/collector/PagingFieldCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.collector;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.FieldDoc;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.Sort;
+import org.opensearch.common.Nullable;
+import org.opensearch.neuralsearch.search.HitsThresholdChecker;
+
+/**
+ * PagingFieldCollector collects the sorted results at the shard level for every individual query
+ * as per search_after criteria applied in the search request.
+ * It collects the list of TopFieldDocs.
+ */
+public final class PagingFieldCollector extends HybridTopFieldDocSortCollector {
+
+    private final Sort sort;
+    private final FieldDoc after;
+
+    public PagingFieldCollector(int numHits, HitsThresholdChecker hitsThresholdChecker, Sort sort, @Nullable FieldDoc after) {
+        super(numHits, hitsThresholdChecker, sort);
+        this.sort = sort;
+        this.after = after;
+    }
+
+    @Override
+    public LeafCollector getLeafCollector(LeafReaderContext context) {
+        docBase = context.docBase;
+        final int afterDoc = after.doc - docBase;
+        return new HybridTopDocSortLeafCollector(sort, after) {
```

Comment:
The logic of this HybridTopDocSortLeafCollector and the one in the SimpleFieldCollector is mostly same. Or at-least we can some part are pretty common, I think we can move some code back to Base class thoughts?

Reply:
No it is not the same, here in the pagingFieldCollector the things which are different 
1) the docBase on which the whole sorting operates.
2) the position of different types of checks like when to check queue[i] is full and whether results are found on previous page or not. 

Lot of if else would be there on the basis of what is sent in the after param of the search request. The code in the common place will become so difficult to understand and maintain.